src/site/content/en/blog/nfc/index.md

```diff
@@ -0,0 +1,911 @@
+  <figcaption class="w-figcaption">Web NFC user prompt</figcaption>
+</figure>
+
+Web NFC is only available to top-level frames, secure browsing contexts (HTTPS
+only), and origins must first request the `"nfc"` [permission] while handling a
+user gesture (e.g a button click). The NDEFReader `scan()` and NDEFWriter
+`write()` methods trigger a user prompt, if access was not previously granted.
+
+```js
+  document.querySelector("#scanButton").onclick = async () => {
+    const reader = new NDEFReader();
+    // Prompt user to allow website to interact with NFC devices.
+    await reader.scan();
+    reader.onreading = event => {
+      // TODO: Handle incoming NDEF messages.
+    };
+  };
+```
+
+The combination of a user-initiated permission prompt and real-world, physical
+movement of bringing the device over a target NFC tag mirrors the chooser
+pattern found in the other file and device-access APIs.
+
+To perform a scan or write, the web page must be visible when the user touches
+an NFC tag with their device. The browser uses haptic feedback to indicate a
+tap. Access to the NFC radio is blocked if the display is off or the device is
+locked. For non visible web pages, receiving and pushing NFC content are
+suspended, and resumed when a web page becomes visible again.
+
+Thanks to the [Page Visibility API], it is possible to track when document
+visibility changes.
+
+```js
+document.onvisibilitychange = event => {
+  if (document.hidden) {
+    // All NFC operations are automatically suspended when document is hidden.
+  } else {
+    // All NFC operations are resumed, if needed.
+  }
+};
+```
+
+## Cookbook {: #cookbook }
+
+Because it is always better to start coding with examples, here are some that
+may help.
+
+### Check for permission
+
+The [Permissions API] allows checking whether the `"nfc"` permission was granted
+previously. This example shows how to scan NFC tags without user interaction if
+access was previously granted, or show a button otherwise. Note that the same
+mechanism works for writing NFC tags as it uses the same permission under the
+hood.
+
+```js
+const reader = new NDEFReader();
+
+async function startScanning() {
+  await reader.scan();
+  reader.onreading = event => {
+    /* handle NDEF messages */
+  };
+}
+
+const nfcPermissionStatus = await navigator.permissions.query({ name: "nfc" });
+if (permissionStatus.state === "granted") {
+  // NFC access was previously granted, so we can start NFC scanning now.
+  startScanning();
+} else {
+  // Show a "scan" button.
+  document.querySelector("#scanButton").style.display = "block";
+  document.querySelector("#scanButton").onclick = event => {
+    // Prompt user to allow UA to send and receive info when they tap NFC devices.
+    startScanning();
+  };
+}
+```
+
+### Abort NFC operations
+
+Using the [AbortController] primitive makes it easy to abort NFC operations and
+other operations that use it. The example below shows you how to pass the
```

Comment:
Delete 'and other operations that use it'.

Reply:
You can use the same abortController for handling NFC operations and other stuff as well (fetching URL, etc.)
Hence why other operations is important here.

Reply:
Interesting, but I don't see how it's relevant to the article. 

By the way, `AbortController` needs code formatting and a link to MDN. 

https://developer.mozilla.org/en-US/docs/Web/API/AbortController

Reply:
I've deleted and formatted properly.